File: src/main/java/me/zeroeightsix/kami/module/modules/render/AntiFog.java
package me.zeroeightsix.kami.module.modules.render;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;

/**
 * Created by 086 on 9/04/2018.
 */
@Module.Info(name = "AntiFog", description = "Disables or reduces fog", category = Module.Category.RENDER)
public class AntiFog extends Module {

    @Setting(name = "Mode") public static VisionMode mode = VisionMode.NOFOG;
    private static AntiFog INSTANCE = new AntiFog();

    public AntiFog() {
        INSTANCE = this;
    }

    public static boolean enabled() {
        return INSTANCE.isEnabled();
    }

    public enum VisionMode {
        NOFOG, AIR
    }

}

File: src/main/java/me/zeroeightsix/kami/module/modules/render/AntiFog.kt
package me.zeroeightsix.kami.module.modules.render

import me.zeroeightsix.kami.module.Category
import me.zeroeightsix.kami.module.Module

/**
 * Created by 086 on 9/04/2018.
 */
internal object AntiFog : Module(
    name = "AntiFog",
    description = "Disables or reduces fog",
    category = Category.RENDER
) {
    private val mode by setting("Mode", VisionMode.NO_FOG)

    private enum class VisionMode {
        NO_FOG, AIR
    }

    val shouldNoFog get() = isActive() && mode == VisionMode.NO_FOG

    val shouldAir get() = isActive() && mode == VisionMode.AIR

    override fun isActive(): Boolean {
        return isEnabled && mc.player != null && mc.player.ticksExisted > 20
    }
}